src/field_theory/separable.lean

```diff
@@ -55,6 +58,113 @@ end
 lemma separable.of_mul_right {f g : polynomial R} (h : (f * g).separable) : g.separable :=
 by { rw mul_comm at h, exact h.of_mul_left }
 
+lemma separable.is_coprime {f g : polynomial R} (h : (f * g).separable) : is_coprime f g :=
+begin
+  have := h.of_mul_left_left, rw derivative_mul at this,
+  exact is_coprime.of_mul_right_right (is_coprime.of_add_mul_left_right this)
+end
+
+theorem separable.of_pow {f : polynomial R} :
+  ∀ {n : ℕ} (h : (f ^ n).separable), is_unit f ∨ (f.separable ∧ n = 1) ∨ n = 0
+| 0     := λ h, or.inr $ or.inr rfl
+| 1     := λ h, or.inr $ or.inl ⟨pow_one f ▸ h, rfl⟩
+| (n+2) := λ h, or.inl $ is_coprime_self.1 h.is_coprime.of_mul_right_left
```

Comment:
I suggest making this one `of_pow'` and then restating `of_pow` where it assumes `\not (is_unit f)` and `n ≠ 0`.

Reply:
Is your `of_pow` more useful?